src/main.ts

```diff
@@ -147,6 +149,94 @@ export default function (): void {
 				}
 			},
 		},
+		{
+			channel: IPCASYNC_EVENTS.GET_ALL_SITES,
+			callback: async () => {
+				// get available providers
+				const providers = await getEnabledBackupProviders();
+
+				// get all repos for each active provider
+				const repos = await Promise.all(
+					providers.map(async (provider: HubProviderRecord) =>
+						await getBackupReposByProviderID(provider.id)));
+
+				// flatten array
+				const allRepos = repos.reduce((acc, curVal) => acc.concat(curVal), []);
+
+				// get all sites for all repos
+				const backupSites = await Promise.all(allRepos.map(async (repo) =>
+					await getBackupSitesByRepoID(repo.hash)));
+
+				// flatten array
+				const allBackupSites = backupSites.reduce((acc, curVal) => acc.concat(curVal), []);
+
+				return allBackupSites;
+			},
+		},
+		{
+			channel: IPCASYNC_EVENTS.GET_REPOS_BY_SITE_ID,
+			callback: async (siteID: number) =>
+				await getBackupReposBySiteID(siteID),
+		},
+		{
+			channel: IPCASYNC_EVENTS.MULTI_MACHINE_GET_AVAILABLE_PROVIDERS,
+			callback: async () => {
+				try {
+					return await getEnabledBackupProviders();
+				} catch (error) {
+					logger.error(`Error - IPCASYNC_EVENTS.MULTI_MACHINE_GET_AVAILABLE_PROVIDERS: ${error.toString()}`);
+					return error;
+				}
+
+			},
+
+		},
+		{
+			channel: IPCASYNC_EVENTS.GET_ALL_SNAPSHOTS,
+			callback: async (siteUUID: string, provider: HubOAuthProviders, requestedPage?: number) => {
+				const repos = await getBackupReposByProviderID(provider);
+
+				const paginationNumber = requestedPage ? requestedPage : 0;
```

Comment:
I think this can be simplified to:

```suggestion
				const paginationNumber = requestedPage ?? 0;
```